workflow-bot-app/src/main/java/com/symphony/bdk/workflow/engine/camunda/UtilityFunctionsMapper.java

```diff
@@ -67,6 +76,41 @@ public static String escape(String s) {
     return new String(JsonStringEncoder.getInstance().quoteAsString(s));
   }
 
+  public static String encodeQueryParameters(String fullUrl) {
+    MultiValueMap<String, String> queryParamsMap =
+        UriComponentsBuilder.fromUriString(fullUrl).build().getQueryParams();
+
+    UriComponentsBuilder uriComponentsBuilder =
+        UriComponentsBuilder.fromUriString(fullUrl);
+
+    queryParamsMap
+        .keySet()
+        .stream()
+        .filter(k -> !isAlreadyEncoded(queryParamsMap.get(k).get(0))).collect(Collectors.toList())
+        .forEach(
+            key -> {
+              try {
+                uriComponentsBuilder.replaceQueryParam(key,
+                    URLEncoder.encode(queryParamsMap.get(key).get(0), StandardCharsets.UTF_8.name()));
+              } catch (UnsupportedEncodingException e) {
+                e.printStackTrace();
+              }
+            });
+
+    return uriComponentsBuilder.build().toUriString();
+  }
+
+  private static boolean isAlreadyEncoded(String value) {
+    boolean isAlreadyEncoded = false;
+    try {
+      isAlreadyEncoded = !URLDecoder.decode(value, StandardCharsets.UTF_8.name()).equals(value);
+    } catch (UnsupportedEncodingException | IllegalArgumentException e) {
+      e.printStackTrace();
```

Comment:
printstacktrace goes to the console so it will not be properly handled by the logging configuration